### ISSUE TITLE
owner of timescaledb schemata is not db owner when DB is created from template1 db

### ISSUE THREAD

**Issue by HeikoOnnebrink** [timescale/timescaledb#605]
**Environment**
PostgreSQL 9.6 
Timescaledb 0.10.0

**Situation 1 : current situation via templatedb1**
timescalcedb is installed and added to shared_prelaod_libraries 
timescaledb extension is added to the template1 db
we have user myuser (nonsuperuser , createdb privs)

user myuser creates database mydb

timescaledb objects are owned by postgres

       List of schemas
        Name          | Owner  
-----------------------+--------
_timescaledb_cache    | postgres
_timescaledb_catalog  | postgres
_timescaledb_internal | postgres
public                | myuser

Problem:
myuser as owner of db mydb is not able to run pg_dump from his db

pg_dump -d mydb -U myuser fails because of lack of privs on timescaledb schemata

**Situation 2:** 
timescalcedb is installed and added to shared_prelaod_libraries 
timescaledb extension is NOT added to the template1 db
we have user myuser (nonsuperuser , createdb privs)

user myuser creates database mydb

we temporarily grant superuser privs to myuser and let him install the timescaledb extension
\c mydb postgres : alter role myuser with superuser;
\c mydb myuser   : creates extension timescaledb;
\c mydb postgres : alter role myuser with nosuperuser;

timescaledb objects are owned by db-owner myuser 

       List of schemas
        Name          | Owner  
-----------------------+--------
_timescaledb_cache    | myuser
_timescaledb_catalog  | myuser
_timescaledb_internal | myuser
public                | myuser

pg_dump -d mydb -U myuser works as myuser is owner of _timescaledb_xx schemata

So basically creating timescaledb via template1 works, but owner of timescaledb schemata is not the db owner but postgres.
Once the user creates the extension the ownership is ok, but here we have the problem that the user must have superuser privs.

Can anything from your side be done in the extension config that the db-owner of the db in which the extension is created is used as owner of the timescaledb schemata?

This would address the root problem and we can add timescaledb directly to our recommended default extensions inside the template1 db.

**Comment by mfreed** [comment]
@HeikoOnnebrink If you might also be willing to check if @svenklemm 's recent PR fixes your problem, that would be great.  Thanks.

**Comment by HeikoOnnebrink** [comment]
@mfreed grant to public could potentially solve the issue as long as it is executed automatically during DB creation from template. Once you have merged the change just let me know from where I can download an updated bin to test the same ..

**Comment by HeikoOnnebrink** [comment]
@mfreed 
we did a short test and simulated a grant to public after creation of a DB from template 1 that incudes the timescaledb extension.
We ended up with this list of privs that need to be granted to allow the DB owner to run a pg_dump against his DB:

grant usage on schema _timescaledb_catalog, _timescaledb_internal, _timescaledb_config, _timescaledb_cache TO PUBLIC;
grant select on all tables in schema _timescaledb_cache to PUBLIC;
grant select on all tables in schema _timescaledb_catalog to PUBLIC;
grant select on all tables in schema _timescaledb_config to PUBLIC;
grant select on all tables in schema _timescaledb_internal to PUBLIC;
grant select on all sequences in schema _timescaledb_config to PUBLIC;
grant select on all sequences in schema _timescaledb_catalog to PUBLIC;
